src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1459 @@
+            **kwargs
+        )
+
+        # Softcopy Presentation LUT
+        if presentation_luts is not None:
+            if presentation_lut_shape is not None:
+                raise TypeError(
+                    'Only one of "presentation_luts" or '
+                    '"presentation_lut_shape" should be provided.'
+                )
+            if len(presentation_luts) == 0:
+                raise ValueError(
+                    'Argument "presentation_luts" must not be empty.'
+                )
+            for v in presentation_luts:
+                if not isinstance(v, LUT):
+                    raise TypeError(
+                        'Items of "presentation_luts" should be of type '
+                        'highdicom.LUT.'
+                    )
+            self.PresentationLUTSequence = presentation_luts
+        else:
+            presentation_lut_shape = (
+                presentation_lut_shape or
+                PresentationLUTShapeValues.IDENTITY
+            )
+            self.PresentationLUTShape = PresentationLUTShapeValues(
+                presentation_lut_shape
+            ).value
+
+
+class PseudoColorSoftcopyPresentationState(_SoftcopyPresentationState):
+
+    """SOP class for a Pseudo-Color Softcopy Presentation State object.
+
+    A Pseudo-Color Softcopy Presentation State object includes instructions for
+    the presentation of a grayscale image as a color image by software.
+
+    """
+
+    def __init__(
+        self,
+        referenced_images: Sequence[Dataset],
+        series_instance_uid: str,
+        series_number: int,
+        sop_instance_uid: str,
+        instance_number: int,
+        manufacturer: str,
+        manufacturer_model_name: str,
+        software_versions: Union[str, Tuple[str]],
+        device_serial_number: str,
+        content_label: str,
+        palette_color_lut: PaletteColorLookupTable,
+        content_description: Optional[str] = None,
+        graphic_annotations: Optional[Sequence[GraphicAnnotation]] = None,
+        graphic_layers: Optional[Sequence[GraphicLayer]] = None,
+        graphic_groups: Optional[Sequence[GraphicGroup]] = None,
+        concept_name_code: Union[Code, CodedConcept, None] = None,
+        institution_name: Optional[str] = None,
+        institutional_department_name: Optional[str] = None,
+        content_creator_name: Optional[Union[str, PersonName]] = None,
+        content_creator_identification: Optional[
+            ContentCreatorIdentificationCodeSequence
+        ] = None,
+        rescale_intercept: Union[int, float, None] = None,
+        rescale_slope: Union[int, float, None] = None,
+        rescale_type: Union[RescaleTypeValues, str, None] = None,
+        modality_lut: Optional[ModalityLUT] = None,
+        copy_modality_lut: bool = False,
+        softcopy_voi_luts: Optional[Sequence[SoftcopyVOILUT]] = None,
+        copy_voi_lut: bool = False,
+        icc_profile: Optional[ImageCmsProfile] = None,
```

Comment:
@dclunie what are the constraints for the ICC Profile in case of a Pseudo-Color Softcopy Presentation State?

The red, blue, and green values specified by the lookup table are obviously in RGB color space. Since an ICC Profile has to be applied, I assume these are considered device-specific rather than standard RGB values.

From a practical perspective, I don't understand why an ICC Profile would be needed. Why would one first apply a Palette Color Lookup Table to colorize pixels and then subsequently correct/adjust the color by applying an ICC Profile? Wouldn't one just use a different Palette Color Lookup Table to begin with?

Are you aware of anyone using ICC Profiles in production to correct the pseudo-color values?